accepted/00000-ptf-repos-in-uyuni.md

```diff
@@ -0,0 +1,101 @@
+- Feature Name: PTF repositories in Uyuni
+- Start Date: 2021-05-17
+
+# Summary
+[summary]: #summary
+
+SUSE want to provide customers PTFs (Product Temporary Fixes) in repositories.
+Uyuni should be able to offer these repositories as channels for syncing and management.
+
+# Motivation
+[motivation]: #motivation
+
+PTFs are Product Temporary Fixes for Service Requests opened by SUSE customers.
+The resulting packages are the last released maintenance update plus a patch which fixes the actual issue.
+
+PTFs are made for dedicated customers. This means every customer has different PTF repositories, with different content and fixes for different service requests.
+
+With Uyuni we should provide the possibility to mirror the content of these repositories and let the customer use it for there clients.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The mapping of SCC repositories to Uyuni Channels is done via a prepared product tree file which contains the definitions for all products and its repositories.
+It enhances the information from SCC with values required for Uyuni.
+
+Such an entry is defined by 3 values:
+
+- the product id
+- the product id of the root product
+- the repository id
+
+The following values are set by the product tree:
+
+- channel label: it must be unique on the whole Uyuni Server
+- parent channel label: what is the parent of this channel
+- channel name: similar to channel label but human readable
+- update tag: a special value to modify the errata id during import
```

Comment:
When SUSE starts an incident which requires a patch, the developer submit the affected package sources to the buildservice. They get build under an (internal) icident id. After QA, when the fix should be released it could happen that multiple update channels are affected. Example "salt":

- salt, salt-minion, ... belongs to sle-module-basesystem15-SP2-Updates
- salt-master, salt-api, ... belongs to sle-module-server-application15-SP2-Updates.

But the `id` in the updateinfo (patch) must be unique. The buildservice has for every release channel an id_template configured which take care for making it unique. The id_templates for this example are defined as:

- `SUSE-SLE-Module-Basesystem-15-SP2-%Y-%C`
- `SUSE-SLE-Module-Server-Applications-15-SP2-%Y-%C`

where `%Y` will be exchanged with the current Year and `%C` is a counter. It is guarantied that the counter is the same for one incident.

When Uyuni now import the  patches from different channels we could import them as single patch. This would result in a lot more patches in the DB as really exists and it could happen that users clone/use only "parts" of a fix if they select only 1 and not all for the action they want to do.

The update tag identify the part of the ID Template which make it unique and try to "rollback" the split made by the buildservice when releasing the patch. The Update Tag in the example case are:

- SLE-Module-Basesystem
- SLE-Module-Server-Applications

During reposync this part is cutted away from the ID. Both patches would end up with the ID `SUSE-15-SP2-%Y-%C`.
reposync search in the DB if an errata with this ID already exists. If yes, it merges the details together. In the DB we have now only 1 errata.

When we re-generate the metadata, we add the update tag again to the resulting updateinfo metadata.

This explains also why the "update tag" must not change. If it would change we cannot find already imported errata and would duplicate everything.

Reply:
Thanks, this clarifies a whole lot.

Can you please copypaste the text above in an appendix section at the end of this file? I would really love not to lose this bit of knowledge, and have a place to refer to when the inevitable questions will come up again!

("Appendix" here means it's not really under discussion as part of the RFC, but brings more context)

Reply:
done